build/tutorials/tutorials-contest/hardhat-fork/README.md

```diff
@@ -0,0 +1,574 @@
+// @ts-ignore
+const pngTokenContract = await ethers.getContractAt("IWAVAX", PNG_ADDRESS)
+//makes sure owner has enough WAVAX balance
+if ((await wavaxTokenContract.balanceOf(account1.address)).lt("1000000000000000000000")) {
+    await wavaxTokenContract.deposit({
+        value: BigNumber.from("1000000000000000000000")
+            .sub(await wavaxTokenContract.balanceOf(account1.address))
+    })
+}
+```
+
+Now we tackle the interesting bit of this test, the actual call to our `Swapper` contract.
+
+We can see that we interact with contract we did not deploy, thanks again to the fork feature. For example here we call
+the approve function of the WAVAX contract. And we also check the balance of our address before the swap.
+
+Then we do the actual call to the swap function of Swapper contract. Passing in the necessary parameters. Here is an
+overview of the parameters :
+
+- `100`, the minimum amount of token we want to receive for this swap. It's intentioanally put low as it is just a test.
+- `[WAVAX_ADDRESS, PNG_ADDRESS]` an array of address that correspond to the path we want to take, basically saying that
+  we want to go from WAVAX to PNG token.
+- `0xd7538cABBf8605BdE1f4901B47B8D42c61DE0367` correspond to the address of the Pangolin Liquidity Pair for WAVAX - PNG.
+- `1000000000` the amount of wavax we are ready to swap for PNG.
+- `1807909162115` this can be ignored as it correspond to the deadline parameter. Which is useless on avalanche (afaik)
+  because transactions are finalized in a very short time frame (< second most of the time).
+
+Then we fetch again the balance of our address. And we check if the balances correspond to our assumptions. If it does,
+it means that our code working as we expect for this functionality.
+
+```ts
+// We tell Wavax contract that we are cool with Swapper contract using our Wavax on our behalve
+await wavaxTokenContract.approve(swapper.address, ethers.constants.MaxUint256);
+
+// Check balance before the swap
+const wavaxBalanceBefore = await wavaxTokenContract.balanceOf(account1.address);
+const pngBalanceBefore = await pngTokenContract.balanceOf(account1.address);
+
+expect(wavaxBalanceBefore).eq("1000000000000000000000");
+expect(pngBalanceBefore).eq(0);
+
+// We call Swapper contract to make a swap from Wavax to Png. I chose some weird values for the swap cause it's just for the sack of this tutorial.
+await swapper.swap(100, [WAVAX_ADDRESS, PNG_ADDRESS], "0xd7538cABBf8605BdE1f4901B47B8D42c61DE0367", 1000000000, 1807909162115);
+
+// Check balance after
+const wavaxBalanceAfter = await wavaxTokenContract.balanceOf(account1.address);
+const pngBalanceAfter = await pngTokenContract.balanceOf(account1.address);
+
+expect(wavaxBalanceAfter).lt(wavaxBalanceBefore);
+expect(pngBalanceAfter).gt(pngBalanceBefore);
+```
+
+If you want to see the code in action, you should run this command in the terminal :
+`npx hardhat test` .
+
+This should produce an output looking like this:
+![img.png](img.png)
```

Comment:
Can you please rename the image file so it is easy to identify where it belongs (BitBook tool we use dumps all the images into a single directory). Simply adding a prefix will do.

Reply:
Done